Sources/NIO/MarkedCircularBuffer.swift

```diff
@@ -18,6 +18,12 @@
 /// writes and mark how far through the buffer the user has flushed, and therefore how far through the buffer is
 /// safe to write.
 public struct MarkedCircularBuffer<E>: CustomStringConvertible, AppendableCollection {
+    #if swift(>=4.2)
+    public typealias RangeType = Range
+    #else
+    public typealias RangeType = CountableRange
```

Comment:
I think the `Type` thing is a Swift 2 leftover (which I personally kinda like, but is inconsistent) and not used at all in Swift 3+ anymore, right?. I would just do:

    #if swift(>=4.2)
    #else
      public typealias Range = CountableRange
    #endif

And use `Swift.Range` in the situations in which it might conflict.

Reply:
@helje5 but in swift < 4.2 there were _distinct_ `Range` and `CountableRange` types, now they're the same (`typealias CountableRange = Range`) and the typealias is deprecated.

But to your point, I could do

```
#if !swift(>=4.2)
public typealias Range = Range
#else
public typealias Range = CountableRange
#endif
```

and then in the methods return `-> Self.Range` ...

Reply:
@Lukasa what do you think? Leave as is (as `RangeType`) or make it `Range`?

Reply:
I'm inclined to call it `RangeType` mostly to avoid the risk of the compiler losing track of what name is meant in a given scope. It'll lead to hard-to-debug errors we don't really need.

Reply:
cool, that was kind of what I thought too. I hope to remove that again with 2.0.0 when we can target 4.2+